server/db/models/users.js

```diff
@@ -1,12 +1,71 @@
 const mongoose = require('mongoose');
```

Comment:
added email field and some validation rules to the model.
pre save function to encrypt the password before saving it to the database.
function to check submitted password matches database hash.
function to login and return the json web token.